app/src/main/kotlin/model/internalGraphs/_DirectedGraph.kt

```diff
@@ -6,9 +6,11 @@ import model.edges.DirectedEdge
 
 abstract class _DirectedGraph<D, E : DirectedEdge<D>> : Graph<D, E>() {
     override fun addEdge(vertex1: Vertex<D>, vertex2: Vertex<D>): E {
-        if (vertex1 == vertex2) throw IllegalArgumentException("Vertices are the same")
+        if (vertex1 == vertex2)
+            throw IllegalArgumentException("Can't add edge from vertex to itself.")
+
```

Comment:
This and some other your changes conflict with formatter, i won't be able to commit if I don't apply formatting for these changes.
I thought that until we have normal formatter configuration we are using ktfmt gradle plugin. At least it fixes something like forgotten whitespaces and indentations.